packages/documentation-redesign/scripts/typedoc-generator.ts

```diff
@@ -0,0 +1,231 @@
+/**
+ * SPDX-FileCopyrightText: 2024 Siemens AG
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import path from 'path';
+import fs from 'fs-extra';
+import Mustache from 'mustache';
+import {
+  Application,
+  IntrinsicType,
+  ReferenceType,
+  TSConfigReader,
+  UnionType,
+} from 'typedoc';
+
+export type TypeDocTarget = {
+  name: string;
+  properties: TypeDocProperty[];
+  type: 'Function' | 'Type';
+  source: string;
+};
+
+export type TypeDocProperty = {
+  name: string;
+  defaultValue?: string;
+  type: string;
+  comment: string;
+  tags: Array<{ tag: string; text?: string }>;
+};
+
+async function generateDocsForEntrypoint(entrypoint: string, targetPath: string) {
```

Comment:
Reduce function nesting as per Sonarqube.
Fn is way too long and complex. Try to extract localized logic to make it more readable.